File: 1921_Eliminate_Maximum_Number_of_Monsters/monster.py
from typing import List

class Solution:
    def eliminateMaximum(self, dist: List[int], speed: List[int]) -> int:
        n=len(dist)
        ans=[]
        for i in range(0,n):
            ans.append((dist[i]+speed[i]-1)//speed[i])

        ans.sort()

        count=0
        for i in range(0,n):
            if(ans[i]<=i):
                break
            count+=1
        
        return count

solution = Solution()

dist = [3,2,4]
speed = [5,3,2]

result = solution.eliminateMaximum(dist, speed)

print(result)

File: 1921_Eliminate_Maximum_Number_of_Monsters/monster.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int eliminateMaximum(vector<int>& dist, vector<int>& speed) {
        int n = dist.size();
        vector<int> ans;

        for (int i = 0; i < n; i++) {
            ans.push_back((dist[i] + speed[i] - 1) / speed[i]);
        }

        sort(ans.begin(), ans.end());

        int count = 0;
        for (int i = 0; i < n; i++) {
            if (ans[i] <= i) {
                break;
            }
            count++;
        }

        return count;
    }
};


int main() {
    vector<int> dist = {3,2,4};
    vector<int> speed = {5,3,2};

    Solution solution;

    int result = solution.eliminateMaximum(dist, speed);

    cout << result << endl;

    return 0;
}